### PR TITLE
Allow customization of ^C with a new color variable

### PR DESCRIPTION
**Pull request by frangio**
## Description

I've modified the `__fish_cancel_commandline` function to print `^C` using a color defined in a new `fish_color_cancel` variable.

I've set the default value of this variable to `-r` to match the current behaviour.

Please let me know if there's anywhere I should add documentation. I looked but didn't find the other color variables documented anywhere.

## TODOs:

- [x] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [ ] User-visible changes noted in CHANGELOG.md

### CODE REVIEW

File: share/functions/__fish_cancel_commandline.fish [hunk 1]
```diff
@@ -3,16 +3,10 @@ function __fish_cancel_commandline
...
-        # mode.
-        #
-        # Set reverse fg/bg color mode, output ^C, restore normal mode, clear to EOL (to erase any
-        # autosuggestion).
+        echo -ns (set_color $fish_color_cancel) "^C" (set_color normal)
         if command -sq tput
-            echo -ns (set_color -r) "^C" (set_color normal) (tput el; or tput ce)
-        else
-            echo -n "^C"
+            # Clear to EOL (to erase any autosuggestion).
```
Comment (krader1961) [inline share/functions/__fish_cancel_commandline.fish]: This is orthogonal to the documented reason for this change and is going to be controversial.

File: share/functions/__fish_cancel_commandline.fish [hunk 2]
```diff
@@ -3,16 +3,10 @@ function __fish_cancel_commandline
...
+            echo -n (tput el; or tput ce)
```
Comment (krader1961) [inline share/functions/__fish_cancel_commandline.fish]: Shouldn't this be `echo -n (tput el ^/dev/null; or tput ce)`? After all, if `el` isn't legal an error is going to be written to stderr.

**Comment by krader1961** [comment]
See the file *doc_src/index.hdr.in* for where the color vars are documented.

File: share/functions/__fish_cancel_commandline.fish (hunk 1 above)
Comment (frangio) [inline share/functions/__fish_cancel_commandline.fish]: Is it possible you misconstrued the diff? This feature was present before too; check out the comment at old line 10.

I just extracted the `^C` echoing out of the conditional statement since it's independent of `tput` availability. It was apparently written that way initially because `tput` was used instead of `set_color`, but it was changed in 1fb92b6.

File: share/functions/__fish_cancel_commandline.fish (hunk 2 above)
Comment (frangio) [inline share/functions/__fish_cancel_commandline.fish]: I think so... I'm not familiar with terminfo, though.

File: share/functions/__fish_cancel_commandline.fish (hunk 1 above)
Comment (krader1961) [inline share/functions/__fish_cancel_commandline.fish]: Yes, you are correct. I was misreading the diff.

**Comment by krader1961** [comment]
Squash merged as commit 349eff0df7. Thanks.
